### PR TITLE
ApacheWebServer

### PR DESCRIPTION
**Pull request by yossifirouz**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/ApacheWebServer/ApacheWebServer/ModelingRules/ApacheWebServer/ApacheWebServerModelingRules.xif [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+| alter message=arrayindex(regextract(_raw_log,"client\s\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\S+[^\:]+\:([^\@]+)"),0)
+
+| alter XDM.Network.event_timestamp= request_time
+| alter XDM.Audit.threat.severity = log_level
+| alter XDM.Audit.identity.uuid=tid 
+| alter XDM.Network.Source.ipv4 = sourceip 
+| alter XDM.Audit.identity.uuid =pid
+| alter XDM.Audit.TriggeredBy.identity.uuid =  error_status_code 
+| alter XDM.Audit.audited_resource.id = Source_code_file_line
+| alter XDM.Audit.threat.description = message
```
Comment (evisochek) [inline Packs/ApacheWebServer/ApacheWebServer/ModelingRules/ApacheWebServer/ApacheWebServerModelingRules.xif:27]: missing ; in the end

File: Packs/ApacheWebServer/ApacheWebServer/ModelingRules/ApacheWebServer/ApacheWebServerModelingRules.xif [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+[MODEL: dataset="apache_httpd_raw", model="Network"]
+
+| filter _raw_log contains "GET" or _raw_log contains "HEAD" or _raw_log contains "POST" or _raw_log contains "DELETE" or _raw_log contains "CONNECT" or _raw_log contains "OPTIONS" or _raw_log contains "TRACE" or _raw_log contains "PATCH"
+ 
+| alter Request_time = arrayindex(regextract(_raw_log , "\[(\d+\/\w+\/\d+\:\d+\:\d+\:\d+\s\S+)\]") , 0) 
+| alter Username = arrayindex(regextract(_raw_log,"(\S+)\s\[\d+\/") ,0)
+| alter requested_line= arrayindex(regextract(_raw_log ,"\"(\w+\s[^\"]+)\"") ,0)
+| alter response_code = arrayindex(regextract(_raw_log ,"\"\s(\d+)\s\d+") ,0)
+| alter User_agent = arrayindex(regextract(_raw_log ,"\"\s\"([^\"]+)") ,0)
+ | alter Referrer= arrayindex(regextract(_raw_log ,"(http[^\"]+)"),0)
```
Comment (evisochek) [inline Packs/ApacheWebServer/ApacheWebServer/ModelingRules/ApacheWebServer/ApacheWebServerModelingRules.xif:40]: Extra whitespace

File: Packs/ApacheWebServer/ApacheWebServer/ModelingRules/ApacheWebServer/ApacheWebServerModelingRules.xif [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+
+| alter XDM.Network.event_timestamp= request_time
+| alter XDM.Network.Source.user.username =username
+| alter XDM.Network.http.url = requested_line
+| alter XDM.Network.http.response_code = response_code 
+| alter  XDM.Network.http.user_agent= User_agent
+| alter XDM.Network.http.referrer =Referrer
+| alter  XDM.Network.Destination.bytes=bytes_size
+| alter XDM.Network.Source.ipv4 =sourceip
```
Comment (evisochek) [inline Packs/ApacheWebServer/ApacheWebServer/ModelingRules/ApacheWebServer/ApacheWebServerModelingRules.xif:57]: missing ; in the end

File: Packs/ApacheWebServer/ApacheWebServer/ModelingRules/ApacheWebServer/ApacheWebServerModelingRules.xif [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+// end extract source_ip
+| alter pid = arrayindex(regextract(_raw_log ,"\[\w+\s(\d+)\:") ,0)
+| alter error_status_code=arrayindex(regextract(_raw_log,"\:\d+\]\s+(\S+)\:") ,0)
+| alter Source_code_file_line=arrayindex(regextract(_raw_log,"tid\s\d+\]\s([^\:|\[]+)") ,0)
+| alter message=arrayindex(regextract(_raw_log,"client\s\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\S+[^\:]+\:([^\@]+)"),0)
+
+| alter XDM.Network.event_timestamp= request_time
+| alter XDM.Audit.threat.severity = log_level
+| alter XDM.Audit.identity.uuid=tid 
+| alter XDM.Network.Source.ipv4 = sourceip
```
Comment (evisochek) [inline Packs/ApacheWebServer/ApacheWebServer/ModelingRules/ApacheWebServer/ApacheWebServerModelingRules.xif:23]: Why there are mapping to network XDM in the Audit data model?

File: Packs/ApacheWebServer/ModelingRules/ApacheWebServerModelingRules.xif [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+| alter response_code = arrayindex(regextract(_raw_log ,"\"\s(\d+)\s\d+") ,0)
+| alter User_agent = arrayindex(regextract(_raw_log ,"\"\s\"([^\"]+)") ,0)
+| alter Referrer= arrayindex(regextract(_raw_log ,"(http[^\"]+)"),0)
+| alter bytes_size=arrayindex(regextract(_raw_log ,"\d\s(\d+)"),0)
+// extract source_ip
+| alter sourceipv4= arrayindex(regextract(_raw_log , "(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}\s)") ,0)
+| alter sourceipv6 = arrayindex(regextract(_raw_log ,"(\w+\:\w+\:\w+\:\w+\:\w+\:\w+\:\w+\:\w+)") ,0)
+| alter sourceip = coalesce(sourceipv4 ,sourceipv6)
+// end extract source_ip
+| alter XDM.Network.event_timestamp= parse_timestamp("%a %b %d %H:%M:%E*S %Y", request_time)
```
Comment (guytamir10) [inline Packs/ApacheWebServer/ModelingRules/ApacheWebServerModelingRules.xif:37]: need to change the parse string to: %d/%b/%Y:%H:%M:%S %Ez
and the request_time to Request_time

**Review by evisochek — APPROVED** [review]

**Pull request merged**